demisto_sdk/commands/common/hook_validations/pre_process_rule.py

```diff
@@ -0,0 +1,111 @@
+import os
+from distutils.version import LooseVersion
+
+from demisto_sdk.commands.common.errors import Errors
+from demisto_sdk.commands.common.hook_validations.content_entity_validator import \
+    ContentEntityValidator
+
+FROM_VERSION_PRE_PROCESS_RULES = '6.5.0'
+
+
+class PreProcessRuleValidator(ContentEntityValidator):
+    def __init__(self, structure_validator=True, ignored_errors=False, print_as_warnings=False,
+                 json_file_path=None, **kwargs):
+        super().__init__(structure_validator, ignored_errors, print_as_warnings,
+                         json_file_path=json_file_path, **kwargs)
+        self.from_version = self.current_file.get('fromServerVersion')
+        self.to_version = self.current_file.get('toServerVersion')
+
+    def is_valid_file(self, validate_rn=True):
+        return self.is_valid_pre_process_rules()
+
+    def is_valid_pre_process_rules(self, validate_rn=True, id_set_file=None, is_ci=False) -> bool:
+        """Check whether the pre_process_rules is valid or not.
+
+        Returns:
+            bool. Whether the pre_process_rules is valid or not
+        """
+        # PreProcessRules files have fromServerVersion instead of fromVersion
+        return all([
+            self.is_valid_version(),
+            self.is_valid_from_server_version(),
+            self.is_script_exist(),
+        ])
+
+    def is_valid_version(self) -> bool:
+        """Checks if version field is valid. uses default method.
+
+        Returns:
+            bool. True if version is valid, else False.
+        """
+        return self._is_valid_version()
+
+    def is_valid_from_server_version(self) -> bool:
+        """Checks if from version field is valid.
+
+        Returns:
+            bool. True if from version field is valid, else False.
+        """
+        if self.from_version:
```

Comment:
if the from version doesn't exist we should also fail the validation.
this can also be obtained by setting the `fromversion` field as required in the schema.

Reply:
The check for is_valid_from_server_version. I do check that it exists. There’s no fromVersion, it’s fromServerVersion.